src/firebase.ts

```diff
@@ -0,0 +1,20 @@
+// Copyright 2020 Cognite AS
+import firebase from 'firebase';
+import { WIDGET_CONF_DB } from 'constants/widgetSettings';
+
+const firebaseConfig = {
+  apiKey: process.env.REACT_APP_API_KEY,
```

Comment:
Since this is loaded to client browser, the app key gets exposed. How do we make sure that someone having access to this key will not be able to access the firebase store?

Reply:
It's not a problem. I'll do this as a best practice. in most cases, these configurations are included in index.html. there is firebase rules for whitelist some domains. only whiteList domains will access data (we can control the access based on the domain).

Reply:
ok, can you please point me to some documentation from google related to this?

Reply:
https://stackoverflow.com/questions/37482366/is-it-safe-to-expose-firebase-apikey-to-the-public

Reply:
and all the firebase examples they put these keys to index.html

Reply:
this is an example project in the firebase site.  https://github.com/FirebaseExtended/firechat/blob/master/website/index.html

Reply:
So apparently this key is only used by firebase to find the api, not for authentication. Can you let me know how your API is protected?

Reply:
since we only use firebase as a database (Firestone) we can't handle authentication things. we can manage authorization using firebase security rules (to some level). [https://firebase.google.com/docs/rules](url)

Reply:
strange. how does it allow protecting different user's resources who are in the same domain?

Reply:
we can manage if the authentication is also in the firebase. but the scenario is different for our case.

Reply:
As discussed, authentication happens through some external auth provider such as google/azure and in firebase we need to provide authorization (Access control). We usually get an access key during authentication and need to find how we can use that to protect the firebase API.

Reply:
As discussed, Lets research and implement auth side of the firebase access possibly using firebase's manual auth flow + ID token returned by cognite SDK under a new task.